packages/cardpay-sdk/sdk/reward-pool/base.ts

```diff
@@ -154,31 +151,55 @@ export default class RewardPool {
     }
     let rewardManager = await getSDK('RewardManager', this.layer2Web3);
     const rule = await rewardManager.getRuleJson(rewardProgramId);
-    json.map((o: any) => {
+    json.data.map(({ attributes: o }: any) => {
       let { validFrom, validTo, token, amount }: FullLeaf = this.decodeLeaf(o.leaf) as FullLeaf;
       const isValid = validFrom <= currentBlock && validTo > currentBlock;
       // filters for known reward tokens
       if (token && rewardTokens.includes(token)) {
-        // filters for proofs has not been claimed
         const explanationTemplate = rewardManager.getClaimExplainer(rule, o.explanationId);
-        if (!knownClaimed) {
-          if (!claimedLeafs.includes(o.leaf)) {
+        if (tokenAddress && tokenAddress == token) {
+          // filters for based upon the tokenAddress if it exists
+          if (!knownClaimed) {
+            // filters for proofs has not been claimed
+            if (!claimedLeafs.includes(o.leaf)) {
+              res.push({
+                ...o,
+                tokenAddress,
+                amount: amount,
+                isValid,
+                explanationTemplate,
+              });
+            }
+          } else {
             res.push({
               ...o,
-              tokenAddress: token,
+              tokenAddress,
               amount: amount,
               isValid,
               explanationTemplate,
             });
           }
         } else {
-          res.push({
-            ...o,
-            tokenAddress: token,
-            amount: amount,
-            isValid,
-            explanationTemplate,
-          });
+          // filters for proofs has not been claimed
+          if (!knownClaimed) {
+            if (!claimedLeafs.includes(o.leaf)) {
+              res.push({
+                ...o,
+                tokenAddress: token,
+                amount: amount,
+                isValid,
+                explanationTemplate,
+              });
+            }
+          } else {
+            res.push({
+              ...o,
+              tokenAddress: token,
+              amount: amount,
+              isValid,
+              explanationTemplate,
+            });
+          }
```

Comment:
I suppose this conditional could be rewritten as something like:
```
if (knownClaimed || !claimedLeafs.includes(o.leaf)) {
   res.push({
     ...o,
     tokenAddress: token,
     amount: amount,
     isValid,
     explanationTemplate,
});
```
Same for the one above.

Reply:
Yes! Thx